### PR TITLE
Cherry-pick: Bug fix packages refreshing (#12433)

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

Cherry-pick https://github.com/DynamoDS/Dynamo/pull/12433 into 2.13

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Cherry-pick https://github.com/DynamoDS/Dynamo/pull/12433 into 2.13

### Reviewers

@QilongTang @filipeotero

### CODE REVIEW

**Review by QilongTang — APPROVED** [review]

**Comment by reddyashish** [comment]
Do not merge this, after resolving the conflicts, something seems to be wrong.

**Comment by reddyashish** [comment]
Closing this PR. Filipe is creating the cherry-pick into 2.13.

**Pull request closed without merging**
